docs/requirements.txt

```diff
@@ -5,3 +5,5 @@ myst-parser==2.0.0
 # Dependencies for extras in `setup.py` needed for documentation.
 matplotlib>=3.0
 lmfit
+# Dependencies needed for building the package
+mypy==0.991
```

Comment:
Small throwback to #581 - it seems that `mypy` is in practice a build dependency currently (due to the need for `stubgen`). I was not sure whether it should be promoted to a proper dependency, so opted for ensuring it becomes a "doc" dependency at least (as this file is installed by readthedocs).